### PR TITLE
actually support merge patches with document streams

### PR DESCRIPTION
**Pull request by BenTheElder**
this explains https://github.com/kubernetes/kubernetes/issues/99802#issuecomment-802997868

### CODE REVIEW

**Review by amwat — COMMENTED** [review]
/lgtm

**Comment by BenTheElder** [comment]
heh, `logging-format` isn't a kubelet flag before 1.19

so this PR is fine, but e2e-k8s.sh needs updating

File: hack/ci/e2e-k8s.sh [hunk 1]
```diff
@@ -64,8 +64,25 @@ build() {
 
 # up a cluster with kind
 create_cluster() {
-  # set KUBELET_LOG_FORMAT=structured to enable structured logging in kubelet
-  KUBELET_LOG_FORMAT="${KUBELET_LOG_FORMAT:-text}"
+  # Grab the version of the cluster we're about to start
```
Comment (BenTheElder) [inline hack/ci/e2e-k8s.sh:67]: I really need to revive the PR to replace this script.

File: hack/ci/e2e-k8s.sh [hunk 2]
```diff
@@ -64,8 +64,25 @@ build() {
...
+  KUBE_VERSION="$(docker run --rm --entrypoint=cat "kindest/node:latest" /kind/version)"
+
+  # Default Log level for all components in test clusters
+  KIND_CLUSTER_LOG_LEVEL=${KIND_CLUSTER_LOG_LEVEL:-4}
+
+  # potentially enable --logging-format
+  kubelet_extra_args="      \"v\": \"${KIND_CLUSTER_LOG_LEVEL}\""
+  if [ -n "${KUBELET_LOG_FORMAT:-}" ]; then
+    case "${KUBE_VERSION}" in
+     v1.1[0-8].*)
```
Comment (aojea) [inline hack/ci/e2e-k8s.sh:77]: how this plays with  versions on CI?

Comment (BenTheElder) [reply hack/ci/e2e-k8s.sh:77]: well < 1.1X we've never supported.
this is actually basically same logic as the ga_only stuff below.

Comment (BenTheElder) [reply hack/ci/e2e-k8s.sh:77]: and we only set KUBELET_LOG_FORMAT in 1.21 / HEAD CI in one job

Comment (aojea) [reply hack/ci/e2e-k8s.sh:77]: scratch it, is used below too, time to :zzz:

**Comment by BenTheElder** [comment]
just a test case flake

**Comment by BenTheElder** [comment]
unblocking structured logging CI. binary change already LGTM

**Pull request merged**